src/javascripts/modules/pills.js

```diff
@@ -1,55 +1,50 @@
 export default class Pills {
   constructor(el) {
     this.el = el
-    this.setVars()
     this.setupListeners()
   }
 
-  setVars() {
-    this.ticking = false
-    this.lastScrollY = 0
-  }
-
   setupListeners() {
     window.addEventListener('load', () => this.handleLoad())
-    window.addEventListener('scroll', () => this.handleScroll())
+    this.createObserver()
   }
 
   handleLoad() {
     this.timeout = setTimeout(() => this.animatePills(), 500)
     this.releaseSqueezeTimeout = setTimeout(() => this.releasePillSqueeze(), 3500)
   }
 
-  handleScroll() {
-    this.lastScrollY = window.scrollY
-    if (!this.ticking) {
-      window.requestAnimationFrame(() => {
-        this.shiftPills()
-      })
+  createObserver() {
+    this.observer
+
+    this.options = {
+      root: null,
+      rootMargin: "20px",
+      threshold: 0,
     }
-    this.ticking = true
+
+    this.observer = new IntersectionObserver(this.handleIntersect, this.options)
+    this.observer.observe(document.querySelector('.green-tablet'))
   }
 
+  handleIntersect(entries, observer) {
+    var titleBox = document.querySelector('.title-box')
```

Comment:
Can you move this query out of the `handleIntersect` callback so that it only runs once? i.e. move to `setVars()`.